File: Flix.Backend/Flix.Services/Database/Clash.cs
using Flix.Model.Enums;
using System.ComponentModel.DataAnnotations;

namespace Flix.Services.Database
{
    public class Clash
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(150)]
        public string Name { get; set; } = string.Empty;

        [Required]
        public string Description { get; set; } = string.Empty;

        public ClashStatus Status { get; set; } = ClashStatus.Upcoming;

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public string? BannerImage { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<ClashEntry> Entries { get; set; } = new List<ClashEntry>();
    }
}

File: Flix.UI/flix_desktop/lib/models/clash.dart
import 'package:flix_desktop/enums/clash_status.dart';
import 'package:json_annotation/json_annotation.dart';

part "clash.g.dart";

@JsonSerializable()
class Clash {
  Clash(
    this.id,
    this.name,
    this.description,
    this.status,
    this.participants,
    this.startDate,
    this.endDate,
    this.createdAt,
    this.bannerImage
  );
  
  
  final int? id;
  final String? name;
  final String? description;
  final ClashStatus? status;
  final int? participants;
  final DateTime? startDate;
  final DateTime? endDate;
  final DateTime? createdAt;
  final String? bannerImage;

  factory Clash.fromJson(Map<String, dynamic> json) => _$ClashFromJson(json);

  Map<String, dynamic> toJson() => _$ClashToJson(this);
}